syntax = "proto3";

message EmptyMessage{}

message PingFilesResponse{
    string ack = 1;
}

message ListFilesResponse{
    repeated string files = 1;
    int32 status = 2;
}

message DownloadFileRequest{
    string fileName = 1;
}

message DownloadFileResponse{
    bytes chunk_data = 1;
}

message UploadFileRequest{
    oneof request {
        string fileName = 1;
        bytes chunk_data = 2;
      }
}

service Files {
    rpc PingFiles(EmptyMessage) returns (PingFilesResponse);
    rpc ListFiles(EmptyMessage) returns (ListFilesResponse);
    rpc DownloadFile(DownloadFileRequest) returns (stream DownloadFileResponse);
    rpc UploadFile(stream UploadFileRequest) returns (EmptyMessage);
}